model_compression_toolkit/core/common/quantization/quantization_params_generation/error_functions.py

```diff
@@ -0,0 +1,149 @@
+import model_compression_toolkit.core.common.quantization.quantization_config as qc
+from model_compression_toolkit.core.common.quantization.quantization_params_generation.kl_selection import \
+    _kl_error_histogram, _kl_error_function
+from model_compression_toolkit.core.common.similarity_analyzer import compute_mse, compute_mae, compute_lp_norm
+from model_compression_toolkit.core.common.target_platform import QuantizationMethod
+
+
+def _mse_error_histogram(q_bins: np.ndarray,
+                         q_count: np.ndarray,
+                         bins: np.ndarray,
+                         counts: np.ndarray) -> np.float:
+    """
+    Compute the error function between a histogram to its quantized version.
+    The error is computed based on the mean square error the distributions have.
+
+    Args:
+        q_bins: Bins values of the quantized histogram.
+        q_count: Bins counts of the quantized histogram.
+        bins: Bins values of the original histogram.
+        counts: Bins counts of the original histogram.
+
+    Returns:
+        MSE between the two histograms.
+    """
+
+    return np.sum((np.power((q_bins - bins)[:-1], 2.0) * counts)) / np.sum(counts)
+
+
+def _mae_error_histogram(q_bins: np.ndarray,
+                         q_count: np.ndarray,
+                         bins: np.ndarray,
+                         counts: np.ndarray) -> np.ndarray:
+    """
+    Compute the error function between a histogram to its quantized version.
+    The error is computed using the mean absolute error between the two histograms.
+
+    Args:
+        q_bins: Bins values of the quantized histogram.
+        q_count: Bins counts of the quantized histogram.
+        bins: Bins values of the original histogram.
+        counts: Bins counts of the original histogram.
+
+    Returns:
+        Mean absolute error of the two histograms.
+    """
+
+    return np.sum((np.abs((q_bins - bins)[:-1]) * counts)) / np.sum(counts)
+
+
+def _lp_error_histogram(q_bins: np.ndarray,
+                        q_count: np.ndarray,
+                        bins: np.ndarray,
+                        counts: np.ndarray,
+                        p: int) -> np.float:
+    """
+    Compute the error function between a histogram to its quantized version.
+    The error is computed based on the distance in Lp-norm between the two distributions.
+    The p-norm to use for the distance computing is passed.
+
+    Args:
+        q_bins: Bins values of the quantized histogram.
+        q_count: Bins counts of the quantized histogram.
+        bins: Bins values of the original histogram.
+        counts: Bins counts of the original histogram.
+        p: p-norm to use for the Lp-norm distance.
+
+    Returns:
+        The Lp-norm distance between the two histograms.
+    """
+
+    return np.sum((np.power(np.abs((q_bins - bins)[:-1]), p) * counts)) / np.sum(counts)
+
+
+def get_threshold_selection_tensor_error_function(quantization_method: QuantizationMethod,
```

Comment:
Is this function only used for Power Of Two? If not rename it to get_tensor_quantization_error_function
(Same for the function below)